h2o-persist-s3/src/test/java/water/persist/PersistS3Test.java

```diff
@@ -57,4 +61,89 @@ public void testS3Import() throws Exception {
       Scope.exit();
     }
   }
+
+  @Test
+  public void testS3UriToKeyUriCredentials() throws Exception {
+    // This test is only runnable in environment with Amazon credentials properly set {AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY}
+    final String accessKey = System.getenv(AWS_ACCESS_KEY_PROPERTY_NAME);
```

Comment:
Not a good idea imho, reasoning please see above ^^^

Reply:
Ideally, the test should also show that the import won't work without the credentials in the URL. When a test depends on an external system (or generally has an external assumption) it is good to validate the assumptions before running the test.